logic/src/androidMain/kotlin/com/wire/kalium/logic/CoreLogic.kt

```diff
@@ -100,6 +100,7 @@ actual class CoreLogic(
                 globalPreferences.value,
                 kaliumConfigs
             ).also {
+                it.keyPackageManager.startObservingKeyPackageCount()
```

Comment:
Would love suggestions for a better place to start the observation. Currently this is logic duplicated in all targets.

Reply:
>I thought starting these in the constructor would be nice enough. But I can see, just like for Sync start/stop, that at some point they must stop. But I question myself: When?

I had them in the constructor initially but then I can't do a `get() = KeyPackageManagerImpl()` in the `UserSessionScope` because then it would never start. So having them in the constructor forces us to initiate all dependencies immediately.

>When should we stop these jobs/async stuff? When the client applications asks us to? Because they went to the background or something? When do we stop?!

I don't we think we need necessarily need to start & stop these jobs, at least not initially. They can stop when the UserSessionScope gets destroyed. It would be future improvement to be able to stop them based certain cases like app going to the background.

Reply:
I'll change it so we start the observing from the constructor with the lazy pattern.

Reply:
Done